engine.js

```diff
@@ -331,7 +397,65 @@ var hmdajson = require('./lib/hmdajson'),
      * -----------------------------------------------------
      */
 
-    // TODO
+    var bindArg = function(arg, objList) {
```

Comment:
This function doesn't really bind an arg, does it? I would expect binding an arg to set an object's property to a value, so this name is misleading. It's more like resolving the value of a property, so the name should fit that.